pkg/platform/api/api.go

```diff
@@ -0,0 +1,219 @@
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, _, _ string) (net.Conn, error) {
+				dialer := net.Dialer{}
+				return dialer.DialContext(ctx, "unix", bottlerocketAPISock)
+			},
+		},
+		// By default, Timeout is set to 0 which would mean no timeout.
+		// Set a 10 second timeout for all requests so we don't wait forever if the API fails to return a response.
+		// The Bottlerocket API should always immediately return a response regardless of the request.
+		// The 10 second value picked here is arbitrary and should be changed if it proves insufficient.
+		Timeout: 10 * time.Second,
+	},
+	}
+}
+
+func (c *apiClient) do(req *http.Request) (*http.Response, error) {
+	var response *http.Response
+	const maxAttempts = 5
+	attempts := 0
+	// Retry up to 5 times in case the Update API is busy; Waiting 10 seconds between each attempt.
+	for ; attempts < maxAttempts; attempts++ {
+		var err error
+		response, err = c.httpClient.Do(req)
+		if err != nil {
+			return nil, errors.Wrapf(err, "update API request error")
+		}
+		if response.StatusCode >= 200 && response.StatusCode < 300 {
+			// Response OK
+			break
+		} else if response.StatusCode == 423 {
+			if attempts < maxAttempts-1 {
+				c.log.Info("API server busy, retrying in 10 seconds ...")
+				// Retry after ten seconds if we get a 423 Locked response (update API busy)
+				time.Sleep(10 * time.Second)
+				continue
+			}
+		}
+		// API response was a non-transient error, bail out.
+		return response, errors.Errorf("bad http response, status code: %d", response.StatusCode)
+	}
+	if attempts == 5 {
+		return nil, errors.New("update API unavailable: retries exhausted")
+	}
+	return response, nil
+}
+
+func (c *apiClient) Get(path string) (*http.Response, error) {
+	req, err := http.NewRequest(http.MethodGet, "http://unix"+path, nil)
+	if err != nil {
+		return nil, err
+	}
+	c.log.WithField("path", path).WithField("method", http.MethodGet).Debugf("update API request")
+	return c.do(req)
+}
+
+func (c *apiClient) Post(path string) (*http.Response, error) {
+	req, err := http.NewRequest(http.MethodPost, "http://unix"+path, http.NoBody)
+	if err != nil {
+		return nil, err
+	}
+	c.log.WithField("path", path).WithField("method", http.MethodPost).Debugf("update API request")
+	return c.do(req)
+}
+
+// GetUpdateStatus returns the update status from the update API
+func (c *apiClient) GetUpdateStatus() (*updateStatus, error) {
+	response, err := c.Get("/updates/status")
+	if err != nil {
+		return nil, err
+	}
+
+	var updateStatus updateStatus
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(body, &updateStatus)
+	if err != nil {
+		return nil, err
+	}
+	return &updateStatus, nil
+}
+
+func (c *apiClient) GetMostRecentCommand() (*commandResult, error) {
+	updateStatus, err := c.GetUpdateStatus()
+	if err != nil {
+		return nil, err
+	}
+	return updateStatus.MostRecentCommand, nil
+}
+
+type oSInfo struct {
```

Comment:
nit: downcase the "term" that you'd normally upcase to export:

```suggestion
type osInfo struct {
```

Reply:
I can't believe I did this :((

Reply:
:rofl: :dog: :watch: :+1: